packages/slate-react/src/hooks/use-children.tsx

```diff
@@ -17,6 +17,68 @@ import {
  * Children.
  */
 
+const Child = (props: {
+  decorations: Range[]
+  parent: Ancestor
+  path: Path
+  child: Descendant
+  isLast: boolean
+  renderElement?: (props: RenderElementProps) => JSX.Element
+  renderPlaceholder: (props: RenderPlaceholderProps) => JSX.Element
+  renderLeaf?: (props: RenderLeafProps) => JSX.Element
+  selection: Range | null
+}) => {
+  const {
+    decorations,
+    parent,
+    path,
+    child,
+    isLast,
+    renderElement,
+    renderPlaceholder,
+    renderLeaf,
+    selection,
+  } = props
+  const decorate = useDecorate()
+  const editor = useSlateStatic()
+
+  const range = Editor.range(editor, path)
+  const sel = selection && Range.intersection(range, selection)
+  const ds = decorate([child, path])
+
+  for (const dec of decorations) {
+    const d = Range.intersection(dec, range)
+
+    if (d) {
+      ds.push(d)
+    }
+  }
+
+  if (Element.isElement(child)) {
+    return (
+      <ElementComponent
+        decorations={ds}
+        element={child}
+        renderElement={renderElement}
+        renderPlaceholder={renderPlaceholder}
+        renderLeaf={renderLeaf}
+        selection={sel}
+      />
+    )
+  } else {
+    return (
+      <TextComponent
+        decorations={ds}
+        isLast={isLast}
+        parent={parent}
+        renderPlaceholder={renderPlaceholder}
+        renderLeaf={renderLeaf}
+        text={child}
+      />
+    )
+  }
```

Comment:
`else` is unnecessary IMO
```suggestion
  return (
    <TextComponent
      decorations={ds}
      isLast={isLast}
      parent={parent}
      renderPlaceholder={renderPlaceholder}
      renderLeaf={renderLeaf}
      text={child}
    />
  )
```

Reply:
I find it clearer without the short-circuit return, but I don't feel strongly about it.